### ISSUE TITLE
Unable to determine "invocation" source in playbook callbacks

### ISSUE THREAD

**Issue by jpmens** [ansible/ansible#1004]
According to https://raw.github.com/ansible/ansible-plugins/master/callbacks/log_plays.py it should be possible to determine the module that invoked a callback by inspecting `invocation`.  

I don't see any mention of an `invocation` object in the result of any task, and particularly not in the result of a `setup`: (latest git version) Is this a bug?
## Playbook

```

---
- hosts: 127.0.0.1
  gather_facts: true
  connection: local
  tasks:
  - name: Ansible says mooooo
    action: command /usr/bin/whoami
```
## ansible-playbook

```
PLAY [127.0.0.1] ********************* 

GATHERING FACTS ********************* 
ok: [127.0.0.1]
{u'verbose_override': True, u'ansible_facts': {u'ansible_machine': u'i386', u'ansible_kernel': u'10.8.0', u'ansible_architecture': u'i386', u'ansible_distribution': u'NA', u'ansible_fqdn': u'jmbp.ww.mens.de', u'ansible_distribution_release': u'NA', u'ansible_python_version': u'2.6.1', u'ansible_system': u'Darwin', u'ansible_hostname': u'jmbp', u'ansible_selinux': False, u'ansible_distribution_version': u'NA'}}
****** INVOCATION [None]

TASK: [Ansible says mooooo] ********************* 
ok: [127.0.0.1]
{u'cmd': [u'/usr/bin/whoami'], u'end': u'2012-09-05 21:10:28.906741', u'stdout': u'jpm', u'changed': True, u'start': u'2012-09-05 21:10:28.902693', u'delta': u'0:00:00.004048', u'stderr': u'', u'rc': 0}
****** INVOCATION [None]

PLAY RECAP ********************* 
127.0.0.1                      : ok=2    changed=1    unreachable=0    failed=0 
```
## callback lib/ansible/callback_plugins/log_plays.py

```
import os
import time
import json

TIME_FORMAT="%b %d %Y %H:%M:%S"
MSG_FORMAT="TIME: %(now)s - CATEG: %(category)s - DATA: %(data)s\n\n"

if not os.path.exists("/tmp/log/ansible/hosts"):
    os.makedirs("/tmp/log/ansible/hosts")

def log(host, category, data):
    print data

    if type(data) == dict:
        invocation = data.pop('invocation', None)
        print "****** INVOCATION [%s]" % invocation
        data = json.dumps(data)

    path = os.path.join("/tmp/log/ansible/hosts", host)
    now = time.strftime(TIME_FORMAT, time.localtime())
    fd = open(path, "a")
    fd.write(MSG_FORMAT % dict(now=now, category=category, data=data))
    fd.close()

class CallbackModule(object):
    def runner_on_ok(self, host, res):
        log(host, 'OK', res)
```
## cat /tmp/log/ansible/hosts/127.0.0.1

```
TIME: Sep 05 2012 21:11:45 - CATEG: OK - DATA: {"verbose_override": true, "ansible_facts": {"ansible_machine": "i386", "ansible_kernel": "10.8.0", "ansible_architecture": "i386", "ansible_distribution": "NA", "ansible_fqdn": "jmbp.ww.mens.de", "ansible_distribution_release": "NA", "ansible_python_version": "2.6.1", "ansible_system": "Darwin", "ansible_hostname": "jmbp", "ansible_selinux": false, "ansible_distribution_version": "NA"}}

TIME: Sep 05 2012 21:11:45 - CATEG: OK - DATA: {"cmd": ["/usr/bin/whoami"], "end": "2012-09-05 21:11:45.797268", "stdout": "jpm", "changed": true, "start": "2012-09-05 21:11:45.791588", "delta": "0:00:00.005680", "stderr": "", "rc": 0}
```

**Comment by jpmens** [comment]
Works with `/usr/bin/ansible`

```
ansible 127.0.0.1 -c local -m setup
127.0.0.1 | success >> {
    "ansible_facts": {
        "ansible_architecture": "i386", 
        "ansible_distribution": "NA", 
        "ansible_distribution_release": "NA", 
        "ansible_distribution_version": "NA", 
        "ansible_fqdn": "jmbp.ww.mens.de", 
        "ansible_hostname": "jmbp", 
        "ansible_kernel": "10.8.0", 
        "ansible_machine": "i386", 
        "ansible_python_version": "2.6.1", 
        "ansible_selinux": false, 
        "ansible_system": "Darwin"
    }, 
    "invocation": {
        "module_args": "", 
        "module_name": "setup"
    }, 
    "verbose_override": true
}

{'invocation': {'module_name': 'setup', 'module_args': u''}, u'verbose_override': True, u'ansible_facts': {u'ansible_machine': u'i386', u'ansible_kernel': u'10.8.0', u'ansible_architecture': u'i386', u'ansible_distribution': u'NA', u'ansible_fqdn': u'jmbp.ww.mens.de', u'ansible_distribution_release': u'NA', u'ansible_python_version': u'2.6.1', u'ansible_system': u'Darwin', u'ansible_hostname': u'jmbp', u'ansible_selinux': False, u'ansible_distribution_version': u'NA'}}
****** INVOCATION [{'module_name': 'setup', 'module_args': u''}]
```

**Comment by mpdehaan** [comment]
ansible --version output please?

**Comment by jpmens** [comment]
`ansible 0.7 (devel ca7b5cc5a3) last updated 2012/09/05 07:45:45 (GMT +200)`

**Comment by mpdehaan** [comment]
Fixed in 96f257c -- which was broken by the code that was trying to hide the invocation data from the CLI.   It's now passed to callbacks.

**Comment by jpmens** [comment]
Tested; works, thank you.
